x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/format_selector.tsx

```diff
@@ -53,20 +65,39 @@ export function FormatSelector(props: FormatSelectorProps) {
     'params' in selectedColumn && selectedColumn.params && 'format' in selectedColumn.params
       ? selectedColumn.params.format
       : undefined;
-  const [state, setState] = useState<State>({
-    decimalPlaces:
-      typeof currentFormat?.params?.decimals === 'number' ? currentFormat.params.decimals : 2,
-  });
 
-  const selectedFormat = currentFormat?.id ? supportedFormats[currentFormat.id] : undefined;
+  const onDecimalsChange = useCallback(
+    (value) => {
+      if (currentFormat?.id) {
+        const validatedValue = Math.min(RANGE_MAX, Math.max(RANGE_MIN, value.decimalPlaces));
+        onChange({
+          id: currentFormat?.id,
+          params: {
+            decimals: validatedValue,
+          },
+        });
+      }
+    },
+    [currentFormat?.id, onChange]
+  );
 
-  const label = i18n.translate('xpack.lens.indexPattern.columnFormatLabel', {
-    defaultMessage: 'Value format',
-  });
+  const initialDecimals = useMemo(
+    () => ({
+      decimalPlaces:
+        typeof currentFormat?.params?.decimals === 'number' ? currentFormat.params.decimals : 2,
+    }),
+    [currentFormat?.params?.decimals]
+  );
```

Comment:
Is there a particular reason, other than legacy, to have the value stored in an object?
Can we store the value as a simple number primitive?